packages/itwin/property-grid/src/components/FilteringPropertyGrid.tsx

```diff
@@ -130,7 +131,7 @@ export const FilteringPropertyGridWithUnifiedSelection = (
     ),
   }), []);
 
-  const { isOverLimit } = usePropertyDataProviderWithUnifiedSelection({ dataProvider: props.dataProvider as IPresentationPropertyDataProvider });
+  const { isOverLimit } = props.disableUnifiedSelection ? { isOverLimit: false } : usePropertyDataProviderWithUnifiedSelection({ dataProvider: props.dataProvider as IPresentationPropertyDataProvider });
```

Comment:
what is the relationship of 'isOverLimit' with 'disableUnifiedSelection'

Reply:
From debugging the code, it looks like unified selection in this case will automatically see which elements are selected in the view and show them in the property grid. The isOverLimit is making sure that the number of elements selected is within the limit to show in the property grid and is what is returned from this hook.

The only place we use disableUnifiedSelection is in the multi-element property grid (I searched GitHub and bentleycs on Azure). It only uses the property grid when displaying a single element that has been selected from the ElementList or when showing the multiple elements (where it shows property values as "varies" etc). 

I can try and figure out a way to make sure it's not over the limit without calling the unified selection hook if needed. But if this hook is called with the multi-element property grid, it breaks it (selecting an element from the ElementList would take you back to the "multiple elements selected" view instead of to the property grid for that specific element, since unified selection will see the multiple elements selected in the view).

Reply:
When disableUnifiedSelection is true, the original property grid code just called VirtualizedPropertyGridWithDataProvider directly (the same thing that happens here if isOverLimit is false). So I think this implementation should be fine.

Reply:
Thanks, if I'm following, we need the filtering in `FilteringPropertyGridUnifiedSelection` that is absent in `VirtualizedPropertyGridWithDataProvider` in order for the null value toggle to work, but currently using `FilteringPropertyGridUnifiedSelection` will trigger the `isOverlimit` clause and not properly show the property grid?

We can't conditionally use a useEffect like `usePropertyDataProviderWithUnifiedSelection` and to me it makes no sense to pass a prop called `disableUnifiedSelection` into a component that has `WithUnifiedSelection` in its name.

If we need a version without the unified selection we need a component called `FilterPropertyGrid` that is just that and this `FilterPropertyGridWithUnifiedSelection` can wrap that with `usePropertyDataProviderWithUnifiedSelection`, then in the PropertyGrid.tsx we can use the component corresponding to whether or not we want unified selection

Reply:
Wouldn't it make more sense to just rename FilteringPropertyGridWithUnifiedSelection to FilteringPropertyGrid and keep the rest as is (passing in disableUnifiedSelection)?

Reply:
Just pushed the change I suggested.

Reply:
This is pushed now